### PR TITLE
[issue_tracker] Update testing plan for issue attachments

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Updated the testing plan for issue_tracker module. In regard to uploading attachments.

- [x] Have you updated related documentation?

#### Link(s) to related issue(s)

* Resolves #  (Reference the issue this fixes, if any.)

#6214

### CODE REVIEW

File: modules/issue_tracker/test/issue_tracker_test_plan.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Issue Tracker Filter Form [Automation Testing]
-1. User can access the page iff they have issue tracker reporter or developer permission
+1. User can access the page if they have issue tracker reporter or developer permission
```
Comment (johnsaigle) [inline modules/issue_tracker/test/issue_tracker_test_plan.md:2]:
```suggestion
1. A user can access the module as long as they have one of the `issue_tracker_reporter` or `issue_tracker_developer` permissions.
```

iff is a short form for "if, and only if,". I agree that it doesn't provide much clarity in this context though so I'm happy to reword the sentence. 

Item 2 should also be reworded.

File: modules/issue_tracker/test/issue_tracker_test_plan.md [hunk 2]
```diff
@@ -24,10 +25,13 @@ Issue Tracker Edit Existing Issue [Manual Testing]
...
 7. Check that watching logging is working - turn it off and on for your current user, and for other watchers on the issue
-
+8. Check uploading new attachment to existing issue.
 
 Permissions [Automation Testing]
 1. Remove access all profile permission.
 2. Remove reporter permission
 3. Remove developer permission
 4. Test that the module behaves correctly as described above. 
+5. Test if user assigned to issue cannot delete attachments of issue owner.
```
Comment (johnsaigle) [inline modules/issue_tracker/test/issue_tracker_test_plan.md:35]: Have steps 5-7 been automated? If not they should not be under this heading of "Automation Testing"

File: modules/issue_tracker/test/issue_tracker_test_plan.md [hunk 3]
```diff
@@ -24,10 +25,13 @@ Issue Tracker Edit Existing Issue [Manual Testing]
 5. Submit just a visit label - this should give an error message.
 6. Check that all values are propagated and saved correctly.
 7. Check that watching logging is working - turn it off and on for your current user, and for other watchers on the issue
-
+8. Check uploading new attachment to existing issue.
```
Comment (johnsaigle) [inline modules/issue_tracker/test/issue_tracker_test_plan.md:28]:
```suggestion
8. Check that an attachment can be adding to an existing issue.
```

File: modules/issue_tracker/test/issue_tracker_test_plan.md [hunk 4]
```diff
@@ -14,7 +14,8 @@ Issue Tracker Create New Issue [Manual Testing]
 4. Submit invalid and valid PSCID and visit label pairs. Error messages should respond accordingly. Not that you cannot submit PSCIDs from other sites unless you have access all profiles permission
 5. Submit just a visit label - this should give an error message.
 6. Check that all values are propagated and saved correctly.
-7. Check that watching logging is working - turn it off and on for your current user, and for other watchers on the issue
+7. Submit with an uploaded attachment should result in success.
```
Comment (johnsaigle) [inline modules/issue_tracker/test/issue_tracker_test_plan.md:17]:
```suggestion
7. Add an attachment to the new issue and make sure that it is successfully uploaded.
```

File: modules/issue_tracker/test/issue_tracker_test_plan.md (hunk 1 above)
Comment (maltheism) [reply modules/issue_tracker/test/issue_tracker_test_plan.md:2]: Wow, I didn't know. Thanks @johnsaigle

File: modules/issue_tracker/test/issue_tracker_test_plan.md (hunk 2 above)
Comment (maltheism) [reply modules/issue_tracker/test/issue_tracker_test_plan.md:35]: @johnsaigle oops mistake.

File: modules/issue_tracker/test/issue_tracker_test_plan.md [hunk 5]
```diff
@@ -24,10 +25,13 @@ Issue Tracker Edit Existing Issue [Manual Testing]
...
+8. Check that an attachment can be adding to an existing issue.
+9. Test if users assigned to issues can upload attachments.
+10. Test if users can delete their own uploaded attachments.
 
 Permissions [Automation Testing]
 1. Remove access all profile permission.
 2. Remove reporter permission
 3. Remove developer permission
 4. Test that the module behaves correctly as described above. 
+5. Test if user assigned to issue cannot delete attachments of issue owner.
```
Comment (johnsaigle) [inline modules/issue_tracker/test/issue_tracker_test_plan.md:37]: Is this automated?

Comment (maltheism) [reply modules/issue_tracker/test/issue_tracker_test_plan.md:37]: thanks @johnsaigle, I somehow didn't remove that.

**Comment by driusan** [comment]
@maltheism this PR is for 23.0 according to the issue label, can you fix the branch?

**Comment by maltheism** [comment]
@driusan okay on it 🙂

File: modules/issue_tracker/test/issue_tracker_test_plan.md [hunk 6]
```diff
@@ -1,5 +1,5 @@
 Issue Tracker Filter Form [Automation Testing]
-1. User can access the page iff they have issue tracker reporter or developer permission
+1. A user can access the module as long as they have one of the `issue_tracker_reporter` or `issue_tracker_developer` permissions.
```
Comment (driusan) [inline modules/issue_tracker/test/issue_tracker_test_plan.md:2]:
```suggestion
1. A user can access the module only if they have one or more of the `issue_tracker_reporter` or `issue_tracker_developer` permissions.
```

**Pull request merged**